com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/utils/FileSystemService.java

```diff
@@ -73,4 +74,15 @@
    */
   @Nonnull
   String decodeURI(@Nonnull String uri);
+
+  /**
+   * This method validate an URI to verify if is defined as extended document, accordining to a
+   * regex.
+   *
+   * @param uri document URI opened in the client
+   * @return true if the document is an extended document, false otherwise
+   */
+  static boolean isFileUnderExtendedSourceFolder(String uri) {
```

Comment:
Are you sure that you want to provide an implementation in the interface? Usually, we must use an interface for other purposes.

Reply:
Well.. this method took in input an URI and this class looks appropriate for that concern and I don't want to create any derivatives of that regex match at the moment, but I am welcome to evaluate alternatives!

Reply:
As you see, we keep implementations in the `WorkspaceFileService` class but use this one only for function declarations. I want to suggest to follow this.